### PR TITLE
fix(appliance): source versions from release registry

### PR DESCRIPTION
**Pull request by craigfurman**
Rather than hardcoding a few. Present the user with versions up to 2
minor revisions back from the version of the appliance itself, which
should be in lock-step with the rest of the monorepo.

One of the last no-changelog PRs to the appliance since we'll likely be publishing an image soon...

Closes https://linear.app/sourcegraph/issue/REL-199/populate-accurate-list-of-versions-to-install
See issue for me talking to myself about a few design trade-offs that are implemented here. Feel free to suggest an alternative approach!

~~Draft until base branch merged, but can be reviewed independently.~~

## Test plan

Admittedly, for E2E, manual testing only, by loading the appliance webpage. Test cases are added for the version logic.

### CODE REVIEW

File: cmd/appliance/README.md [hunk 1]
```diff
@@ -11,3 +11,34 @@ Appliance runs as a standard Kubernetes Deployment and utilizes Kubernetes [cont
 ## Own
 
 For more information or for help, see the [Release Team](https://handbook.sourcegraph.com/departments/engineering/teams/release/).
+
+## Development
```
Comment (Chickensoupwithrice) [inline cmd/appliance/README.md:15]: Love to see it ❤️

File: internal/appliance/versions.go [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+		nminor = append(nminor, version)
+	}
+
+	// If we have not collected n minor versions, we tolerate one major version
+	// transition and search for the remaining (n - whatever) minor version
+	// matches there.
+	if len(nminor) == 0 {
+		return nil, errors.Newf("found no versions within %d minor revisions of %s", n, latestSupportedVersion)
+	}
+	if latestSupported.Minor()-nminor[0].Minor() < 2 {
```
Comment (Chickensoupwithrice) [inline internal/appliance/versions.go:54]: Should this be 
`if latestSupported.Minor()-nminor[0].Minor() < n {` 
`n` instead of `2`?

**Review by Chickensoupwithrice — APPROVED** [review]
Minor nits that shouldn't block the merge :)

File: internal/appliance/versions.go (hunk 2 above)
Comment (craigfurman) [reply internal/appliance/versions.go:54]: Oof, good catch! It's probably a bit overengineering of me to even parameterise this, not just hardcode 2 inside the function, but 🤷 

will fix then merge

**Pull request merged**